### PR TITLE
tools:  improve formatting of weekly report

### PR DESCRIPTION
**Pull request by dhellmann**
- Use the filename of the enhancement document being changed in each PR to add grouping information to the summary line.
- Remove "enhancement:" prefix from titles.
- Use the PR description as the summary if the PR doesn't contain an enhancement document.

[sample output](https://hackmd.io/RdANRydbT3SKUQQCiH8O4g)

### CODE REVIEW

File: tools/enhancements/summary.go [hunk 1]
```diff
@@ -100,6 +100,53 @@ func extractSummary(body string) string {
 	return b.String()
 }
 
+// getEnhancementFilename returns the filename of the enhancement
+// document modified by the pull request
+func getEnhancementFilename(pr int) (filename string, err error) {
```
Comment (russellb) [inline tools/enhancements/summary.go:105]: not using this anymore, i think?  want to keep it anyway?

Comment (dhellmann) [reply tools/enhancements/summary.go]: I went ahead and removed it

Comment (dhellmann) [reply tools/enhancements/summary.go]: I also added a Makefile to be able to run the linter locally.

**Pull request merged**